File: Backend/omechoo/app/schemas/requests/menu.py
from typing import Any
from pydantic import BaseModel, Field


class MenuRecommendRequest(BaseModel):
    """메뉴 추천 요청"""
    
    included_categories: list[str] | None = Field(
        None,
        description="포함할 카테고리 목록 (없으면 전체)",
        examples=["korean", "western"]
    )
    excluded_categories: list[str] | None = Field(
        None,
        description="제외할 카테고리 목록",
        examples=["japanese"]
    )
    attributes: dict[str, Any] | None = Field(
        None,
        description="메뉴 속성 필터 (예: {'main_base': 'rice', 'spiciness': 2})",
        examples=[{"main_base": "rice", "spiciness": 2}]
    )
    limit: int = Field(
        default=5,
        ge=1,
        le=10,
        description="추천 개수 (1-10)"
    )

File: Backend/omechoo/frontend/src/types/menu.ts
import type { ApiResponse } from './common';

export type MenuCategory = 
  | 'korean' 
  | 'chinese' 
  | 'japanese' 
  | 'western' 
  | 'asian' 
  | 'cafe' 
  | 'fast_food' 
  | 'fusion' 
  | 'buffet' 
  | 'other';

export type MainBase = 
  | 'rice' 
  | 'noodle' 
  | 'bread' 
  | 'meat' 
  | 'seafood' 
  | 'vegetable' 
  | 'etc';

export type Temperature = 'hot' | 'cold' | 'neutral';

export type Heaviness = 1 | 2 | 3; // Light(1), Medium(2), Heavy(3)

export interface MenuRecommendRequest {
  included_categories?: MenuCategory[];
  excluded_categories?: MenuCategory[];
  attributes?: {
    main_base?: MainBase | MainBase[];
    spiciness?: number;
    temperature?: Temperature;
    heaviness?: Heaviness;
  };
  limit?: number;
}

export interface Menu {
  id: string;
  name: string;
  category: string;
  description?: string;
  main_base?: string;
  spiciness?: number;
  temperature?: string;
  heaviness?: number;
  tags?: string[];
  search_keywords?: string[];
}

export type MenuRecommendResponse = ApiResponse<Menu[]>;